### PR TITLE
 Additions and fixes to documentation 

### PR DESCRIPTION
**Pull request by hakonsbm**
Some updates, additions, and adjustments to the documentation, from @stinebuu and me. Please review it to make sure that we haven't introduced any erroneous information.

There are still some left to update, but we are unsure about what to do with these, so we will handle them in the main PR.

### CODE REVIEW

File: nestkernel/connection_manager.cpp [hunk 1]
```diff
@@ -975,22 +971,6 @@ nest::ConnectionManager::data_connect_single( const index source_id,
...
- * If source is a gid and target is a list of GIDs then the sources is
- * connected to all targets.
- * If source is a list of GIDs and target is a GID, then all sources are
- * connected to the target.
- * At this stage, the task of connect is to separate the dictionary into one
- * for each thread and then to forward the connect call to the connectors who
- * can then deal with the details of the connection.
- *
- * @note This method is used only by DataConnect.
- */
```
Comment (jakobj) [inline nestkernel/connection_manager.cpp]: did you remove this comment completely or move it somewhere else?

File: nestkernel/send_buffer_position.h [hunk 2]
```diff
@@ -54,11 +54,32 @@ class SendBufferPosition
...
+  /**
+   * Returns whether the part of the buffer on the specified rank has been
+   * filled.
+   *
+   * @param rank Rank denoting which part of the buffer we check
+   */
   bool is_chunk_filled( const thread rank ) const;
+
+  /**
+   * Returns whether the entire buffer has been filled.
```
Comment (jakobj) [inline nestkernel/send_buffer_position.h]: this actually just checks whether all parts of the MPI buffer assigned to this thread are filled, not the whole MPI buffer

File: nestkernel/connection_manager.cpp (hunk 1 above)
Comment (hakonsbm) [reply nestkernel/connection_manager.cpp]: It was almost a duplication of what was already in the h-file, I just moved `@note`. And it's better to generally keep documentation in the h-files anyway.

File: nestkernel/send_buffer_position.h (hunk 2 above)
Comment (hakonsbm) [reply nestkernel/send_buffer_position.h]: Fixed!

**Pull request merged**